### PR TITLE
develop: add non-transactional delete

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by ekexium — APPROVED** [review]

File: develop/dev-guide-delete-data.md [hunk 1]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
 
-</SimpleTab>
\ No newline at end of file
+</SimpleTab>
+
+## 非事务批量删除
+
+> **注意：**
+>
+> TiDB 从 v6.1.0 版本开始支持[非事务 DML 语句](/non-transactional-dml.md)特性。在更低版本上无法使用此特性。
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:307]:
```suggestion
> TiDB 从 v6.1.0 版本开始支持[非事务 DML 语句](/non-transactional-dml.md)特性。在 TiDB v6.1.0 以下版本上无法使用此特性。
```

File: develop/dev-guide-delete-data.md [hunk 2]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
+
+### 非事务批量删除使用前提
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:309]:
```suggestion
### 使用前提
```

File: develop/dev-guide-delete-data.md [hunk 3]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
+
+请先行**仔细**阅读[非事务 DML 语句](/non-transactional-dml.md)一章。非事务批量删除，本质是以牺牲事务的原子性、隔离性为代价，增强批量数据处理场景下的性能和易用性。
+
+因此在使用过程中，需要极为小心，否则，因为操作的非事务特性，在误操作时会导致严重的后果（如数据丢失等）。
+
+### 非事务批量删除 SQL 语法
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:315]:
```suggestion
### SQL 语法
```

File: develop/dev-guide-delete-data.md [hunk 4]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
+
+|    参数    |      描述      |
+| :--------: | :------------: |
+| `{dividing_column}`  |      非事务批量删除的划分列      |
+| `{batch_size}` | 非事务批量删除的每批大小 |
+| `{delete_statement}` | 删除语句 |
+
+此处仅展示非事务批量删除的简单用法，详细文档可参考 TiDB 的[非事务 DML 语句](/non-transactional-dml.md)。
+
+### 非事务批量删除使用示例
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:333]:
```suggestion
### 使用示例
```

File: develop/dev-guide-delete-data.md [hunk 5]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
+
+## 非事务批量删除
+
+> **注意：**
+>
+> TiDB 从 v6.1.0 版本开始支持[非事务 DML 语句](/non-transactional-dml.md)特性。在更低版本上无法使用此特性。
+
+### 非事务批量删除使用前提
+
+请先行**仔细**阅读[非事务 DML 语句](/non-transactional-dml.md)一章。非事务批量删除，本质是以牺牲事务的原子性、隔离性为代价，增强批量数据处理场景下的性能和易用性。
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:311]:
```suggestion
在使用非事物批量删除前，请先**仔细**阅读[非事务 DML 语句](/non-transactional-dml.md)。非事务批量删除，本质是以牺牲事务的原子性、隔离性为代价，增强批量数据处理场景下的性能和易用性。
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: develop/dev-guide-delete-data.md [hunk 6]
```diff
@@ -41,6 +41,7 @@ DELETE FROM {table} WHERE {filter}
 - 需要删除大量行(数万或更多)的时候，使用[批量删除](#批量删除)，这是因为 TiDB 单个事务大小限制为 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)（默认为 100MB）。
 - 如果你需要删除表内的所有数据，请勿使用 `DELETE` 语句，而应该使用 [TRUNCATE](/sql-statements/sql-statement-truncate.md) 语句。
 - 查看 [性能注意事项](#性能注意事项)。
+- 如果你的删除无需事务性，可使用[非事务批量删除](#非事务批量删除)，来提高性能。
```
Comment (SunRunAway) [inline develop/dev-guide-delete-data.md:44]: 在客户要删除大量数据的时候，我们应该主力推荐“非事务批量删除”。

Comment (Icemap) [reply develop/dev-guide-delete-data.md:44]: 可是无事务的话，误操作的代价太高了。我这里给出一个 Trade off 给用户吧。

File: develop/dev-guide-delete-data.md (hunk 3 above)
Comment (Icemap) [reply develop/dev-guide-delete-data.md:315]: 这篇文章中已经存在一个 SQL 语法的标题了。

File: develop/dev-guide-delete-data.md (hunk 4 above)
Comment (Icemap) [reply develop/dev-guide-delete-data.md:333]: ditto

File: develop/dev-guide-delete-data.md (hunk 3 above)
Comment (Icemap) [reply develop/dev-guide-delete-data.md:315]: 所以在 `非事务批量删除` 这一节下的所有二级标题都加了这个前缀。

File: develop/dev-guide-delete-data.md [hunk 7]
```diff
@@ -41,6 +41,7 @@ DELETE FROM {table} WHERE {filter}
 - 需要删除大量行(数万或更多)的时候，使用[批量删除](#批量删除)，这是因为 TiDB 单个事务大小限制为 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)（默认为 100MB）。
 - 如果你需要删除表内的所有数据，请勿使用 `DELETE` 语句，而应该使用 [TRUNCATE](/sql-statements/sql-statement-truncate.md) 语句。
 - 查看 [性能注意事项](#性能注意事项)。
+- 在需要大批量删除数据的场景下，[非事务批量删除](#非事务批量删除)对性能的提升十分明显。但与之相对的，这将丢失删除的事务性，因此***无法***进行回滚，请务必正确进行操作选择。
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:44]:
```suggestion
- 在需要大批量删除数据的场景下，[非事务批量删除](#非事务批量删除)对性能的提升十分明显。但与之相对的，这将丢失删除的事务性，因此**无法**进行回滚，请务必正确进行操作选择。
```

File: develop/dev-guide-delete-data.md [hunk 8]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
 
-</SimpleTab>
\ No newline at end of file
+</SimpleTab>
+
+## 非事务批量删除
+
+> **注意：**
+>
+> TiDB 从 v6.1.0 版本开始支持[非事务 DML 语句](/non-transactional-dml.md)特性。在 TiDB v6.1.0 以下版本上无法使用此特性。
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:307]:
```suggestion
> TiDB 从 v6.1.0 版本开始支持[非事务 DML 语句](/non-transactional-dml.md)特性。在 TiDB v6.1.0 以下版本中无法使用此特性。
```

**Review by Oreoxmt — APPROVED** [review]

File: develop/dev-guide-delete-data.md [hunk 9]
```diff
@@ -297,4 +298,42 @@ func deleteBatch(db *sql.DB, startTime, endTime time.Time) (int64, error) {
...
+| `{delete_statement}` | 删除语句 |
+
+此处仅展示非事务批量删除的简单用法，详细文档可参考 TiDB 的[非事务 DML 语句](/non-transactional-dml.md)。
+
+### 非事务批量删除使用示例
+
+以上方[批量删除例子](#批量删除例子)场景为例，可使用以下 SQL 语句进行非事务批量删除：
+
+```sql
+BATCH ON `rated_at` LIMIT 1000 DELETE FROM `ratings` WHERE `rated_at` >= "2022-04-15 00:00:00" AND  `rated_at` <= "2022-04-15 00:15:00";
```
Comment (Oreoxmt) [inline develop/dev-guide-delete-data.md:338]:
```suggestion
BATCH ON `rated_at` LIMIT 1000 DELETE FROM `ratings` WHERE `rated_at` >= "2022-04-15 00:00:00" AND `rated_at` <= "2022-04-15 00:15:00";
```

**Comment by Oreoxmt** [comment]
@SunRunAway PTAL again

**Pull request merged**